### PR TITLE
Add huggingface hub versions

### PR DESCRIPTION
**Pull request by aurianer**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-huggingface-hub/package.py [hunk 1]
```diff
@@ -16,6 +16,23 @@ class PyHuggingfaceHub(PythonPackage):
...
+    version("0.22.0", sha256="304f1e235c68c0a9f58bced47f13d6df241a5b4e3678f4981aa1e4f4bce63f6d")
+    version("0.21.4", sha256="e1f4968c93726565a80edf6dc309763c7b546d0cfe79aa221206034d50155531")
+    version("0.21.3", sha256="26a15b604e4fc7bad37c467b76456543ec849386cbca9cd7e1e135f53e500423")
+    version("0.21.2", sha256="839f2fc69fc51797b76dcffa7edbf7fb1150176f74cb1dc2d87ca00e5e0b5611")
+    version("0.21.1", sha256="c458ae6b3e8e197472c4ef01d8cc5f8b3ddb70e9288afcd494753d832dac3a70")
+    version("0.21.0", sha256="b11200deefaa802f3f5060e4689bf2a441d65b6fbf9325c27cb631ec9dd7eb68")
+    version("0.20.3", sha256="94e7f8e074475fbc67d6a71957b678e1b4a74ff1b64a644fd6cbb83da962d05d")
+    version("0.20.2", sha256="215c5fceff631030c7a3d19ba7b588921c908b3f21eef31d160ebc245b200ff6")
+    version("0.20.1", sha256="8c88c4c3c8853e22f2dfb4d84c3d493f4e1af52fb3856a90e1eeddcf191ddbb1")
+    version("0.20.0", sha256="3b41e9e1cd8342d5e0b63f1921b6dbd03222215d2d4874f032a465421394e001")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-huggingface-hub/package.py]: Are all these versions really necessary, or can we just stick to the latest patch for each minor?

Comment (alalazo) [reply var/spack/repos/builtin/packages/py-huggingface-hub/package.py]: Also, did you check if really nothing changed in the deps constraints?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-huggingface-hub/package.py (hunk 1 above)
Comment (aurianer) [reply var/spack/repos/builtin/packages/py-huggingface-hub/package.py]: Right now we only need the latest version, so I can just add this one and I will add others progressively if needed.

Comment (aurianer) [reply var/spack/repos/builtin/packages/py-huggingface-hub/package.py]: I didn't check the dependencies manually but I checked that it installs

Comment (alalazo) [reply var/spack/repos/builtin/packages/py-huggingface-hub/package.py]: > Right now we only need the latest version, so I can just add this one and I will add others progressively if needed.

Yeah, let's do that

**Comment by aurianer** [comment]
Updated, sorry for the delay

**Review by alalazo — APPROVED** [review]

**Pull request merged**